parlai/scripts/generate_model_card.py

```diff
@@ -0,0 +1,1628 @@
+        '--evaluation-report-file',
+        '-eval-rf',
+        type=str,
+        default=None,
+        help="evaluation report file",
+    )
+    gmc.add_argument(
+        '--extra-args-path',
+        '-exargs',
+        type=str,
+        default=None,
+        help='path to .json file with extra arguments used for different stages of report generation and later for quantitative analyses section generation; please do NOT use the shortened format (ie. t=<task>); check documentation for more info',
+    )
+    gmc.add_argument(
+        '--quantitative-report-files',
+        '-quant-rfs',
+        type=str,
+        default=[],
+        nargs='*',
+        help='quantitative report file (with different subgroups); if multiple, please separate with comma, and (optional) also add a field in the report file stating what kind of subgroup it is; note that this is only applicable for classifier type models',
+    )
+    gmc.add_argument(
+        '--include-misc',
+        type='bool',
+        default=True,
+        help='whether to include the miscellaneous dropdown (fields that were not included in other dropdowns); by default, the value is True.',
+    )
+    gmc.add_argument(
+        '--quant-metrics',
+        type=str,
+        default=[],
+        help='Other metrics to include in the quantitative analysis',
+    )
+    return parser
+
+
+def decide_model_type(opt, model_dict):
+    """
+    based on key words in the model dictionary, we try to figure out what type of model
+    this is.
+
+    Note: can later try to make it more accurate by tracing the model's ancestors
+    """
+    # decide from user input
+    if opt.get('model_type'):
+        return opt['model_type']
+
+    # fields to check and key words that match to a model type
+    check_fields = ('agent', 'title', 'path', 'description')
+    key_words = {
+        'ranker': RANKER,
+        'classifier': CLASSIFIER,
+        'generator': GENERATOR,
+        'retrieval': RETRIEVER,
+        'retriever': RETRIEVER,
+    }
+
+    # decide from model_dict
+    for key, model_type in key_words.items():
+        for field in check_fields:
+            if model_dict.get(field) and key in model_dict.get(field):
+                return model_type
+
+
+###################################
+# Script for generating model card
+###################################
+
+
+@register_script('generate_model_card', aliases=['gmc'])
+class GenerateModelCard(ParlaiScript):
+    @classmethod
+    def setup_args(cls):
+        return setup_args()
+
+    def run(self):
+        self.opt.log()
+        self.verbose = self.opt['verbose']
+        self.mode = self.opt['mode'].split(':')[0]
+
+        if self.mode not in {M_gen, M_edit, M_final}:
+            raise RuntimeError('The mode' + self.mode + 'was unrecognized.')
+
+        self.general_setup()
+        if self.mode == M_gen:
+            self._set_evaltask()
+            jobs, args = self._gen_jobs()
+            self.save_reports(jobs, args)
```

Comment:
made it get the jobs needed to run instead of having it run everything all the time